clang/include/clang/Basic/AttrDocs.td

```diff
@@ -2503,26 +2503,61 @@ device kernel, the attribute is not ignored and it is propagated to the kernel.
     [[intel::num_simd_work_items(N)]] void operator()() const {}
   };
 
-If the`` intel::reqd_work_group_size`` or ``cl::reqd_work_group_size``
-attribute is specified on a declaration along with a
-intel::num_simd_work_items attribute, the work group size attribute
-argument (the first argument) must be evenly divisible by the argument specified
-in the ``intel::num_simd_work_items`` attribute.
+The arguments to reqd_work_group_size are ordered based on which index
+increments the fastest. In OpenCL, the first argument is the index that
+increments the fastest, and in SYCL, the last argument is the index that
+increments the fastest.
+
+In OpenCL, all three arguments are required.
+
+In SYCL, the attribute accepts either one, two, or three arguments; in each
+form, the last (or only) argument is the index that increments fastest.
+The number of arguments passed to the attribute must match the dimensionality
+of the kernel the attribute is applied to.
+
+If the reqd_work_group_size attribute is specified on a declaration along
+with ``intel::num_simd_work_items``, the required work group size specified
+by ``intel::num_simd_work_items`` attribute must evenly divide the index that
+increments fastest in the reqd_work_group_size attribute.
 
 .. code-block:: c++
 
   struct func {
     [[intel::num_simd_work_items(4)]]
-    [[intel::reqd_work_group_size(64, 64, 64)]]
+    [[intel::reqd_work_group_size(3, 4, 64)]]
     void operator()() const {}
   };
 
   struct bar {
-    [[intel::reqd_work_group_size(64, 64, 64)]]
+    [[intel::reqd_work_group_size(3, 4, 64)]]
     [[intel::num_simd_work_items(4)]]
     void operator()() const {}
   };
 
+  [[cl::reqd_work_group_size(7, 10, 20)]]
+  [[intel::num_simd_work_items(5)]] void fun2() {}
+
+  [[intel::num_simd_work_items(4)]]
+  [[cl::reqd_work_group_size(5, 4, 8)]] void fun3() {}
+
+  struct func1 {
+    [[intel::num_simd_work_items(4)]]
+    [[cl::reqd_work_group_size(5, 4, 64)]]
+    void operator()() const {}
+  };
+
+  struct bar1 {
+    [[cl::reqd_work_group_size(3, 4, 64)]]
+    [[intel::num_simd_work_items(4)]]
+    void operator()() const {}
+  };
+
+  [[intel::num_simd_work_items(2)]]
+  __attribute__((reqd_work_group_size(4, 2, 3))) void test();
```

Comment:
Then on this one you can add a comment like: `// Note, '4' is evenly divisible by '2'; in OpenCL, the first argument to the attribute is the one which increments fastest.`